src/tdastro/sources/redback_models.py

```diff
@@ -0,0 +1,200 @@
+      * redshift - The object's redshift. [from PhysicalModel]
+      * t0 - The t0 of the zero phase, date. [from PhysicalModel]
+    Additional parameterized values are used for specific redback models.
+
+    References
+    ----------
+    * redback - https://ui.adsabs.harvard.edu/abs/2024MNRAS.531.1203S/abstract
+    * Individual models might require citation. See references in the redback documentation.
+
+    Attributes
+    ----------
+    source : function
+        The underlying source function that maps time + wavelength to flux.
+    source_name : str
+        The name used to set the source.
+    source_param_names : list
+        A list of the source model's parameters that we need to set.
+
+    Parameters
+    ----------
+    source : str or function
+        The name of the redback model function used to generate the SEDs or
+        the actual function itself.
+    parameters : dict, optional
+        A dictionary of parameter setters to pass to the source function.
+    **kwargs : dict, optional
+        Any additional keyword arguments.
+    """
+
+    # A class variable for the units so we are not computing them each time.
+    _FLAM_UNIT = uu.erg / uu.second / uu.cm**2 / uu.AA
+
+    def __init__(
+        self,
+        source,
+        *,
+        parameters=None,
+        **kwargs,
+    ):
+        # Check that the parameters passed in the dictionary and keyword arguments
+        # do not overlap, so we only have one source of truth. This is needed for
+        # parameters like `redshift` that overlap core parameters.
+        if parameters is None:
+            parameters = {}
+        for key in parameters:
+            if key in kwargs:
+                raise ValueError(
+                    f"Parameter '{key}' specified in both the parameters dictionary "
+                    "and as a parameter itself. Please include it only in the dictionary."
+                )
+
+        super().__init__(**kwargs)
+
+        # Use the parameter dictionary to create settable parameters for the model.
+        # Some of these might have already been added by the superclass's constructor,
+        # so we just change it.
+        self.source_param_names = []
+        for key, value in parameters.items():
+            if key in self.setters:
+                self.set_parameter(key, value)
+            else:
+                self.add_parameter(key, value)
+            self.source_param_names.append(key)
+
+        # Create the source itself.
+        if isinstance(source, str):
+            try:
+                import redback
+            except ImportError as err:
+                raise ImportError(
+                    "redback package is not installed be default. To use the RedbackWrapperModel, "
+                    "please install redback. For example, you can install it with "
+                    "`pip install redback` or `conda install conda-forge::redback`."
```

Comment:
It is not on conda yet https://anaconda.org/search?q=redback

```suggestion
                    "`pip install redback`."
```